packages/firebase_ml/android/src/main/java/io/flutter/plugins/firebaseml/ModelManager.java

```diff
@@ -0,0 +1,132 @@
+    }
+    if (conditionsToMap.get("requireDeviceIdle")) {
+      conditionsBuilder.requireDeviceIdle();
+    }
+    if (conditionsToMap.get("requireWifi")) {
+      conditionsBuilder.requireWifi();
+    }
+
+    FirebaseModelDownloadConditions conditions = conditionsBuilder.build();
+    FirebaseModelManager.getInstance()
+        .download(remoteModel, conditions)
+        .addOnSuccessListener(
+            new OnSuccessListener<Void>() {
+              @Override
+              public void onSuccess(Void v) {
+                result.success(remoteModelToMap(remoteModel));
+              }
+            })
+        .addOnFailureListener(
+            new OnFailureListener() {
+              @Override
+              public void onFailure(@NonNull Exception exception) {
+                result.error("FirebaseModelManager", exception.getLocalizedMessage(), null);
+              }
+            });
+  }
+
+  private static void getLatestModelFile(@NonNull MethodCall call, @NonNull final Result result) {
+    assert (call.argument("modelName") != null);
+    String modelName = call.argument("modelName");
+
+    final FirebaseCustomRemoteModel remoteModel =
+        new FirebaseCustomRemoteModel.Builder(modelName).build();
+
+    FirebaseModelManager.getInstance()
+        .getLatestModelFile(remoteModel)
+        .addOnCompleteListener(
+            new OnCompleteListener<File>() {
+              @Override
+              public void onComplete(@NonNull Task<File> task) {
+                File modelFile = task.getResult();
+                if (modelFile != null) {
+                  result.success(modelFile.getAbsolutePath());
+                } else {
+                  result.error(
+                      "FirebaseModelManager", task.getException().getLocalizedMessage(), null);
+                }
+              }
+            });
+  }
+
+  private static void isModelDownloaded(@NonNull MethodCall call, @NonNull final Result result) {
+    assert (call.argument("modelName") != null);
+    String modelName = call.argument("modelName");
+
+    final FirebaseCustomRemoteModel remoteModel =
+        new FirebaseCustomRemoteModel.Builder(modelName).build();
+
+    FirebaseModelManager.getInstance()
+        .isModelDownloaded(remoteModel)
+        .addOnCompleteListener(
+            new OnCompleteListener<Boolean>() {
+              @Override
+              public void onComplete(@NonNull Task<Boolean> task) {
+                Boolean isModelDownloaded = task.getResult();
+                if (isModelDownloaded != null) {
+                  result.success(isModelDownloaded);
+                } else {
+                  result.error(
+                      "FirebaseModelManager", task.getException().getLocalizedMessage(), null);
+                }
+              }
+            });
+  }
+
+  private static Map<String, String> remoteModelToMap(FirebaseCustomRemoteModel model) {
+    Map remoteModelToMap = new HashMap<String, String>();
```

Comment:
Should define the types of the Map's generics to avoid unchecked calls. Can also used inferred generics when instantiating. ie:
Map<String, String> remoteModelToMap = new HashMap<>();

Reply:
Done